### PR TITLE
Mlp

### PR DESCRIPTION
**Pull request by erwityk**
Attempts to solve the Kaggle project using an artificial neural network with a back-propagation algorithm.  This will be merged into `master` if we determine it to be the best model that we will upload to Kaggle for scoring.

To dos:

- [ ] Determine the relevant model parameters from the input data.
- [ ] Clean the training and prediction data, including any necessary feature scaling or dummy variable creation
- [ ] Implement code to train the network, and output survival predictions as a .csv file.
- [ ] Determine the relevant model hyperparameters (number of hidden layers, number of neurons, etc.)  This will likely involve some amount of testing with the training dataset.
- [ ] Create the final survival predictions.

### CODE REVIEW

File: prep.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+`pclass` is the ticket class (1st, 2nd, or 3rd), which should be transformed
+into two dummy variables.
+`sex` should be transformed into a dummy variable
+`age` should be normalized
+`sibsp` should be normalized
+`parch` should be normalized
+`ticket` is an ID variable that is not relevant for the model
+`fare` should be normalized
+`cabin` [[[potentially relevant, but not sure how to deal with this just yet]]]
+`embarked` [[[not sure if relevant]]]
```
Comment (erwityk) [inline prep.py:16]: We need to figure out if `cabin` and `embarked` should be used for the model, and how.

**Comment by erwityk** [comment]
Just pushed a function in mlp.py that allows us to play around with the hidden layers in the model.  So far I'm getting around 80% accuracy, which I don't think is particularly great, but I haven't gotten this to a point where all the features are included/properly cleaned.
